geometry/optimization/dev/BUILD.bazel

```diff
@@ -12,20 +12,7 @@ load(
 )
 load("//tools/lint:lint.bzl", "add_lint_tests")
 
-drake_cc_library(
-    name = "cspace_free_polytope",
-    srcs = ["cspace_free_polytope.cc"],
-    hdrs = ["cspace_free_polytope.h"],
-    deps = [
-        "//common/symbolic:monomial_util",
-        "//geometry/optimization:c_iris_collision_geometry",
-        "//geometry/optimization:c_iris_separating_plane",
-        "//solvers:choose_best_solver",
-        "//solvers:mathematical_program",
-        "//solvers:mosek_solver",
-        "//solvers:solve",
-    ],
-)
+exports_files(["cspace_free_polytope.cc"])
```

Comment:
FYI This is where things went wrong; a `dev/BUILD.bazel` file must always have narrow visibility.  Exporting to `visibility:public` broke through the firewall.